### DISCUSSION TITLE
Unit test ValueResolver and mocking the ResolutionContext parameter

### DISCUSSION THREAD

**Discussion by gdoron** [LuckyPennySoftware/AutoMapper#3726] [NA]
I have a custom ValueResolver which I want to unit test.

```csharp
public class CreateByAdminResolver : IValueResolver<BaseManagementTenantEntity, BaseManagementTenantDto, AdminDto> {
    public AdminDto Resolve(BaseManagementTenantEntity source, BaseManagementTenantDto destination,
        AdminDto destMember,
        ResolutionContext context) {

        if (!context.Items.TryGetValue("constValue", out var usersObject))
            return null;

        var users = (List<UserDto>)usersObject;
        // do something with users...
        return somethingCool;
    }
}
```
Problem is, the `Resolve` method gets `ResolutionContext` instead of `IResolutionContext`, so it's impossible to mock.
And it has `internal` constructors so it's even impossible to just "new" as was suggested by @jbogard long time ago in #761, (probably the class was changed completely since 2015...)

So, how can we Unit Test and mock the `ResolutionContext` today? 

Best,
Doron

**Comment by lbargaoanu** [comment]
https://docs.automapper.org/en/latest/10.0-Upgrade-Guide.html#the-resolutioncontext-no-longer-has-a-public-constructor

**Reply by gdoron** [reply]
@lbargaoanu thanks for the reply!
A. I'm wondering on the motivation though, not to provide a public constructor or make `Resolve` method get an `IResolutionContext` instead of a class.

B. Are you opened for a PR on changing the `IValueResolver ` interface to get an interface?

**Reply by lbargaoanu** [reply]
No, there is no need to mock the context, you can just have the real thing.

**Reply by jbogard** [reply]
Exactly this would be like mocking StringBuilder, it's just an object.
